control/beacon/store.go

```diff
@@ -47,7 +48,17 @@ func (c chainCheckerOption) apply(o *storeOptions) {
 // beacons are verifiable with cryptographic material available in the local
 // trust store.
 func WithCheckChain(p ChainProvider) StoreOption {
```

Comment:
`WithCheckChain()` is not used anymore, should we maintain it or can we drop it? Beside, do we want to allow the double-wrapping:
```
beacon.NewBeaconStore(policies, db,
    beacon.WithSelectionAlgorithm(beacon.NewChainsAvailableAlgo(p, algo)),
    beacon.WithCheckChain(p),
)
```
Just asking, perhaps with another selection algorithm it makes sense...